### ISSUE TITLE
Help with compatibility problems with the dataset

### ISSUE THREAD

**Issue by AndreaSDG** [shap/shap#2740]
### Discussed in https://github.com/slundberg/shap/discussions/2739

<div type='discussions-op-text'>

<sup>Originally posted by **AndreaSDG** October 26, 2022</sup>
So it's been a while I'm working on my thesis that talks about explainability using the Seagate multivariate time-series dataset about the HDD manufacturing process and I was trying to use SHAP to understand more about my model. I used keras to build its  structure:

model = tf.keras.Sequential()
    model.add(layers.Conv1D(128, 2, input_shape=(X_train.shape[1], X_train.shape[2]), dilation_rate=2, padding='causal'))
    model.add(layers.Conv1D(128, 2, dilation_rate=2, padding='causal'))
    model.add(layers.LSTM(128, return_sequences=True, recurrent_dropout = 0.5, dropout = 0.5))
    model.add(layers.LSTM(128, return_sequences=True, recurrent_dropout = 0.5, dropout = 0.5))
    model.add(layers.LSTM(128, return_sequences=True, recurrent_dropout = 0.5, dropout = 0.5))
    model.add(layers.LSTM(128, return_sequences=False, recurrent_dropout = 0.5, dropout = 0.5))
    model.add(layers.Dense(1, activation='sigmoid'))

So it's a binary classification problem to detect if an HDD is failed or not by analyzing very small multivariate time-series (2 timesteps for each sample), I have a 3D input (samples, time-steps, features). This is the dataset information overall:

------ TRAIN ------

X_train shape:  (2191, 2, 817)
y_train shape:  (2191, 1)
Negative labels:  [1748]
Positive labels:  [443]

------ VALIDATION ------

X_val shape:  (235, 2, 817)
y_val shape:  (235, 1)
Negative labels:  [196]
Positive labels:  [39]

------ TEST ------

X_test shape:  (983, 2, 817)
y_test shape:  (983, 1)
Negative labels:  [2183]
Positive labels:  [8]

So my plan was to use a couple of SHAP plots, at least the main ones like force_plot and summary_plot but It seems I'm not able to make them work with this kind of dataset due to its shape. I started creating the explainer by using:

explainer = shap.DeepExplainer(model, X_train)
shap_values = explainer.shap_values(sample) # Just one sample of the dataset because SHAP required a lot of time just for 50 samples

 I also searched for people with the same problem with multivariate time-series but no progress overall. So I thought I could've asked some help here, thanks in advance for any kind of help!!
</div>

**Comment by detrin** [comment]
@AndreaSDG Could you provide the dataset or sample of it so that I can execute your code and debug it?

> So my plan was to use a couple of SHAP plots, at least the main ones like force_plot and summary_plot but It seems I'm not able to make them work with this kind of dataset due to its shape. I started creating the explainer by using:

> explainer = shap.DeepExplainer(model, X_train)
shap_values = explainer.shap_values(sample) # Just one sample of the dataset because SHAP required a lot of time just for 50 samples

Posting the stack error would be helpful.
